nussl/separation/composite/overlap_add.py

```diff
@@ -0,0 +1,89 @@
+from librosa.core import audio
+from .. import SeparationBase
+from ... import AudioSignal
+import numpy as np
+import warnings
+import matplotlib.pyplot as plt
+
+class OverlapAdd(SeparationBase):
+    def __init__(self, separation_object, window_length=15, find_permutation=False):
+        """[summary]
+
+        Parameters
+        ----------
+        separation_object : SeparationBase
+            Separation object that overlap and add is applied to.
+        window_length : int, optional
+            Window length of overlap/add window, by default 15 seconds.
+            The hop length will be half the window length.
+        find_permutation : bool, optional
+            Whether or not to find the permutation between chunks before combining.
+        """
+        super().__init__(separation_object.audio_signal)
+
+        self.hop_length = window_length / 2
+        self.window_length = window_length
+        self.separation_object = separation_object
+        self.find_permutation = find_permutation 
+
+    def run(self, *args, **kwargs):
+        audio_signal = self.separation_object.audio_signal
+
+        if audio_signal.signal_duration <= self.window_length:
+            estimates = self.separation_object()
+            self.audio = np.stack([e.audio_data for e in estimates], axis=-1)
+            warnings.warn("Signal duration is less than the window length. "
+                          "Running algorithm on entire signal and returning.")
+            return self.audio
+        
+        win_samples = int(self.window_length * audio_signal.sample_rate)
+        hop_samples = int(self.hop_length * audio_signal.sample_rate)
+        audio_signal.zero_pad(hop_samples, hop_samples)
+        num_samples = audio_signal.signal_length
+        
+        audio_data = np.zeros_like(audio_signal.audio_data)
+        chunk_starts = np.arange(0, num_samples - hop_samples, hop_samples)
+        estimates_from_chunks = []
+        
+        for i, start_idx in enumerate(chunk_starts):
+            end_idx = start_idx + win_samples
+            window = AudioSignal.get_window('hanning', win_samples)[None, :]
```

Comment:
No way to set the window type?

Reply:
Hanning is the only perceptually valid one for overlap/add, here.